exercises/hamming/test/test_hamming.c

```diff
@@ -14,95 +14,67 @@ static void test_empty_strands(void)
    TEST_ASSERT_EQUAL(0, compute("", ""));
 }
 
-static void test_rejects_null_strand(void)
+static void test_single_identical_strands(void)
 {
    TEST_IGNORE();               // delete this line to run test
-   TEST_ASSERT_EQUAL(-1, compute(NULL, "A"));
-}
-
-static void test_rejects_other_null_strand(void)
-{
-   TEST_IGNORE();
-   TEST_ASSERT_EQUAL(-1, compute("A", NULL));
-}
-
-static void test_no_difference_between_identical_strands(void)
-{
-   TEST_IGNORE();
    TEST_ASSERT_EQUAL(0, compute("A", "A"));
 }
 
-static void test_identical_long_strands(void)
-{
-   TEST_IGNORE();
-   TEST_ASSERT_EQUAL(0, compute("GGACTGA", "GGACTGA"));
-}
-
-static void test_hamming_distance_for_single_nucleotide_strand(void)
+static void test_single_letter_different_strands(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(1, compute("A", "G"));
+   TEST_ASSERT_EQUAL(1, compute("G", "T"));
 }
 
-static void test_complete_hamming_distance_for_small_strand(void)
+static void test_long_identical_strands(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(2, compute("AG", "CT"));
+   TEST_ASSERT_EQUAL(0, compute("GGACTGAAATCTG", "GGACTGAAATCTG"));
 }
 
-static void test_small_hamming_distance(void)
+static void test_long_different_strands(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(1, compute("AT", "CT"));
-}
-
-static void test_small_hamming_distance_in_longer_strand(void)
-{
-   TEST_IGNORE();
-   TEST_ASSERT_EQUAL(1, compute("GGACG", "GGTCG"));
+   TEST_ASSERT_EQUAL(9, compute("GGACGGATTCTG", "AGGACGGATTCT"));
 }
 
-static void test_rejects_extra_length_on_first_strand_when_longer(void)
+static void test_disallow_first_strand_when_longer(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(-1, compute("AAAG", "AAA"));
+   TEST_ASSERT_EQUAL(-1, compute("AATG", "AAA"));
 }
 
-static void test_rejects_extra_length_on_other_strand_when_longer(void)
+static void test_disallow_second_strand_when_longer(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(-1, compute("AAA", "AAAG"));
+   TEST_ASSERT_EQUAL(-1, compute("ATA", "AGTG"));
 }
 
-static void test_large_hamming_distance(void)
+static void test_disallow_left_empty_strand(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(4, compute("GATACA", "GCATAA"));
+   TEST_ASSERT_EQUAL(-1, compute(NULL, "G"));
 }
 
-static void test_hamming_distance_in_very_long_strand(void)
+static void test_disallow_right_empty_strand(void)
 {
    TEST_IGNORE();
-   TEST_ASSERT_EQUAL(9, compute("GGACGGATTCTG", "AGGACGGATTCT"));
+   TEST_ASSERT_EQUAL(-1, compute("G", NULL));
 }
```

Comment:
Well, now I'm having second thoughts about my earlier comment regarding `""` and `NULL`. I don't think that the spec makes much sense here. The name and the check don't match. An empty strand (`""`) shouldn't be an error , but a missing strand (`NULL`) should be. Further confusing things is that because the strand lengths don't match, it would be an error regardless of whether it was empty or missing so it's confusing to say that an _empty_ strand should be treated as an error.

I think these tests would make far more sense if named `test_disallow_<which>_strand_missing`.

Reply:
Agreed, the logic means that these two tests don't make sense.
I've removed them and set the related UUIDs to false in the TOML file.
Have also opened an issue on the problem specs repo about it (https://github.com/exercism/problem-specifications/issues/1761).

Reply:
Thanks! 👍